tests/phpunit/CRM/Mailing/BaseMailingSystemTest.php

```diff
@@ -315,11 +307,10 @@ public function urlTrackingExamples() {
       array('url_tracking' => 1),
     );
     $cases[] = array(
-      // Plain-text URL's are tracked in plain-text emails...
-      // but not in HTML emails.
+      // Plain-text URLs are not tracked.
       "<p>Please go to: http://example.net/</p>",
       ";<p>Please go to: http://example\.net/</p>;",
-      ';Please go to: .*extern/url.php\?u=\d+&qid=\d+;',
+      ';Please go to: http://example\.net/;',
```

Comment:
This seems like a significant policy change...

Reply:
Mixed feelings about disabling click-tracking on plain-text mailings:
 * On text mailings, you'll probably get more recipients clicking the link if the URL looks sensible. I'd guess that a majority of users would prefer to enable click-tracking on HTML but disable click-tracking on text.
 * I don't think *all* users would, and I think the change is a little big for a point-release. I'd rather see some kind of toggle to allow smoother transition.
 * I'm very much open to improving FlexMailer to make this easier to manipulate programmatically. (It should actually be possible already by modifying or replacing the `'civi_flexmailer_text_click_tracker`, but it's a little obscure -- the [docs](https://docs.civicrm.org/flexmailer/en/latest/develop/#services) mention the service but describe how to replace it. That's something like `$container->findDefinition('civi_flexmailer_text_click_tracker')->setMethodCalls(...'myfunc'...)`. I'd be make a prettier DX.)